### PR TITLE
[DELETE] /api/exams/{examId} #276

### PR DESCRIPTION
**Pull request by timt1028**
Resolve:#276

### CODE REVIEW

File: Domain/Academy/src/main/java/tw/waterball/judgegirl/academy/domain/repositories/ExamRepository.java [hunk 1]
```diff
@@ -45,4 +45,6 @@
     int countAnswersInQuestion(Question.Id id, int studentId);
 
     boolean hasStudentParticipatedExam(int studentId, int examId);
```
Comment (Johnny850807) [inline Domain/Academy/src/main/java/tw/waterball/judgegirl/academy/domain/repositories/ExamRepository.java:47]: rename this to `isExaminee`

File: Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java [hunk 2]
```diff
@@ -557,6 +556,17 @@ void testAnswerQuestionConcurrentlyWithSubmissionQuotas() throws Exception {
         });
     }
 
+    @Test
+    void GivenExamSavedAndGetExam_WhenDeleteExamById_ShouldExamBeNotEmpty() throws Exception {
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java:560]: bad name, the testcase's name should be **narrative**

please improve it yourself for a while

File: Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java [hunk 3]
```diff
@@ -557,6 +556,17 @@ void testAnswerQuestionConcurrentlyWithSubmissionQuotas() throws Exception {
...
+        var exam = createExamAndGet(new Date(), new Date(), "exam");
+
+        deleteExam(exam.id);
+
+        Optional<Exam> optionalExam = examRepository.findById( exam.id);
+        assertTrue(optionalExam.isEmpty());
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java:566]: `assertTrue(examRepository.findById(exam.id).isEmpty())`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done

File: Spring-Boot/Spring-Boot-Academy/src/main/java/tw/waterball/judgegirl/springboot/academy/repositories/JpaExamRepository.java [hunk 4]
```diff
@@ -158,8 +158,13 @@ public int countAnswersInQuestion(Question.Id id, int studentId) {
...
 
     @Override
-    public boolean hasStudentParticipatedExam(int studentId, int examId) {
+    public boolean isExaminee(int studentId, int examId) {
         return jpaExamineeDAO.existsById_StudentIdAndId_ExamId(studentId, examId);
     }
 
+    @Override
+    public void deleteExamById(int examId) {
+        jpaExamDAO.deleteById(examId);
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Academy/src/main/java/tw/waterball/judgegirl/springboot/academy/repositories/JpaExamRepository.java:167]:
```Java
        try {
            jpaExamDAO.deleteById(examId);
        } catch (EmptyResultDataAccessException ignored) {
        }
```

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**